src/hooks/core/post/usePostHook.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {Context} from '../../../context';
+import {setMainFeeds, setTopicFeeds} from '../../../context/actions/feeds';
+import {followUserAnon, setFollow, setUnFollow, unfollowUserAnon} from '../../../service/profile';
+import removePrefixTopic from '../../../utils/topics/removePrefixTopic';
+import TopicPageStorage from '../../../utils/storage/custom/topicPageStorage';
+
+const usePostHook = () => {
+  const [profileContext] = (React.useContext(Context) as any).profile;
+  const [feedsContext, feedsContextDispatch] = (React.useContext(Context) as any).feeds;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `as any` to cast the context is a potential code smell. It would be better to define an interface for your context that includes `profile` and `feeds`, and use that interface when calling `useContext`. This will provide type safety and make it easier to understand what data is expected in the context.

```diff
- const [profileContext] = (React.useContext(Context) as any).profile;
- const [feedsContext, feedsContextDispatch] = (React.useContext(Context) as any).feeds;

+ interface AppContext {
+   profile: [ProfileContext, Function];
+   feeds: [FeedsContext, Function];
+ }
+ 
+ const { profile: [profileContext], feeds: [feedsContext, feedsContextDispatch] } = React.useContext<AppContext>(Context);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->